mutex.go

```diff
@@ -108,27 +102,37 @@ func (m *mutex) Unlock() {
 			if err == nil {
 				res.Body.Close()
 				if res.StatusCode == 204 {
-					return
+					return nil
 				}
 			}
 		}
-		time.Sleep(time.Duration(i) * time.Millisecond)
+		select {
+		case <-time.After(time.Duration(i) * time.Millisecond):
+			continue
+		case <-ctx.Done():
+			return errors.New("unlock request timed out")
+		}
 	}
 }
 
+// httpClient is overwritten in tests
+var httpClient = func(ctx context.Context) (*http.Client, error) {
+	const scope = "https://www.googleapis.com/auth/devstorage.full_control"
+	return google.DefaultClient(ctx, scope)
+}
+
 // New creates a GCS-based sync.Locker.
 // It uses Application Default Credentials to make authenticated requests
 // to Google Cloud Storage. See the DefaultClient function of the
 // golang.org/x/oauth2/google package for App Default Credentials details.
 //
 // If ctx argument is nil, context.Background is used.
 //
-func New(ctx context.Context, project, bucket, object string) (sync.Locker, error) {
+func New(ctx context.Context, project, bucket, object string) (*mutex, error) {
```

Comment:
No need for the `project` argument at all, right.
It can also be removed from the `mutex` struct.